cpp/src/toppra/constraint.cpp

```diff
@@ -33,46 +33,64 @@ void allocateLinearPart(std::size_t N, Eigen::Index k, Eigen::Index m,
     F.resize(N);
     g.resize(N);
   }
-  for (Vector& x : a) x.resize(m);
-  for (Vector& x : b) x.resize(m);
-  for (Vector& x : c) x.resize(m);
-  for (Matrix& x : F) x.resize(k, m);
-  for (Vector& x : g) x.resize(k);
+  for (Vector& x : a) { x.resize(m);
+}
+  for (Vector& x : b) { x.resize(m);
+}
+  for (Vector& x : c) { x.resize(m);
+}
+  for (Matrix& x : F) { x.resize(k, m);
+}
+  for (Vector& x : g) { x.resize(k);
+}
```

Comment:
Is it a linter who introduced those changes ?
I find the syntax very misleading. I have nothing against adding brackets. I can understand
```cpp
for (Vector& x : a) {
  x.resize(m);
}
```
```cpp
for (Vector& x : a) { x.resize(m); }
```
but the current result, I cannot understand the motivation for it.

Reply:
And more important, I don't understand why the closing bracket is not indented as the for.

Reply:
This commit is from the linter. Do you suggest to change all to the below? I can make that change.

```
for (Vector& x : a) { x.resize(m); }
```

Reply:
You mean you will change the linter rules or you will manually do it ?
There are many of them and, if you change them manually, the linter will complain again. In some project in my lab, we use clang-format, with google-style guide. It does not completely match my taste but I have nothing better to propose (EDIT there are some other proposed here https://clang.llvm.org/docs/ClangFormatStyleOptions.html).

Reply:
I use clang-tidy, which I suppose only works on the logic level. Style-wise it doesn't care. Will eventually add a config for clang-format and enforce it. 

I don't have strong opinion on this matter. As long as the code is nicely formatted and consistent I am ok.

Reply:
`for (Vector& x : a) { x.resize(m); }` is fine for me.
Mind that there are plenty of places where clang-tidy introduced a badly indented closing bracket.

I let you choose whether or not to use clang-format. I usually don't see it as necessary. I can see that we did not follow the same convention in many places (not restricted for formatting) anyway, so I guess formatting does not make much sense.

Reply:
I think we should have a consistent format style, and clang-format is the easiest way to do it. 

The only problem with introducing clang-format now is merge conflicts, so I will not format the code you are working on. However, I will run that on code I am working on.